code/game/mecha/equipment/weapons/weapons.dm

```diff
@@ -391,6 +391,35 @@
 /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/missile_rack/flashbang/clusterbang/limited/rearm()
 	return//Extra bit of security
 
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/missile_rack/cleaner
+	equip_cooldown = 6 SECONDS
+	name = "\improper N23 Rotary Janitation Launcher"
+	icon_state = "mecha_grenadelnchr"
+	origin_tech = "combat=4;engineering=4"
+	projectile = /obj/item/grenade/chem_grenade/cleaner
+	fire_sound = 'sound/effects/bang.ogg'
+	projectiles = 6
+	missile_speed = 1.5
+	projectile_energy_cost = 1000
+	var/det_time = 20
+	size = 1
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/missile_rack/cleaner/action(target, params)
+	if(!action_checks(target))
+		return
+	set_ready_state(0)
+	var/obj/item/grenade/chem_grenade/cleaner/grenade = new projectile(chassis.loc)
+	playsound(chassis, fire_sound, 50, TRUE)
+	grenade.throw_at(target, missile_range, missile_speed)
+	projectiles--
+	log_message("Fired from [name], targeting [target].")
+	spawn(det_time)
```

Comment:
Please use a timer here instead of spawning. This should translate very cleanly.